pmd-core/src/main/java/net/sourceforge/pmd/Report.java

```diff
@@ -474,39 +477,89 @@ public boolean treeIsEmpty() {
      * Returns an iteration over the reported violations.
      *
      * @return an iterator
+     *
+     * @deprecated The {@link ReportTree} is deprecated
      */
+    @Deprecated
     public Iterator<RuleViolation> treeIterator() {
         return violationTree.iterator();
     }
 
+    /**
+     * @deprecated Use {@link #getViolations()}
+     */
+    @Deprecated
     @Override
     public Iterator<RuleViolation> iterator() {
         return violations.iterator();
     }
 
+
+    /**
+     * Returns an unmodifiable list of violations that were suppressed.
+     */
+    public List<SuppressedViolation> getSuppressedRuleViolations() {
```

Comment:
The getter for the violations is only called getViolations and not getRuleViolations. In that sense, I think, this method should be called `getSuppressedViolations()`.

If you agree, we need to create the new method `getSuppressedViolations()` and deprecated the existing `getSuppressedRuleViolations()`.

Reply:
All of this sounds good. Will do